src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,73 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        cache_key = f"r.c.spc:{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
+            " ", ""
+        )
+        session_count = cache.get(cache_key)
+        if session_count is None:
+            filters = {"project_id": [event.project_id]}
+            if environment_id:
+                filters["environment"] = [environment_id]
+            result_totals = raw_query(
```

Comment:
I feel like we should actually be querying the last hour of data and then dividing this value. So if we have a 1 minute rule, query the last hour, then / 60. If we have a 5 minute rule, / 12, etc. This should help us avoid variability from sessions coming in late.

Reply:
Oof, I wrote a response to this yesterday but apparently didn't post it and now it's lost.

This is an interesting idea that I could see actually helping. It could potentially miss spikes though? It seems like it'd have the effect of firing an alert too early (if sessions spike the calculation will use a lower session value than it would today, making the % higher than maybe it should be), and the other side is if sessions drop we might actually "miss" an alert (the % would be lower due to the higher session count averaged over the last hour).

In practice, I don't expect sessions to fluctuate in the same way errors do. I'm happy to try this...we can maybe try less time too (30 minutes). What do you think of following up with this while we're able to poke around as-is with the condition in production?

Reply:
The main problem here is that sessions arrive a little late, so fetching sessions for the last minute is unlikely to be accurate. I think at the very least, you'd need to fetch them from the last 5-10 minutes.

Since we're just testing this for now, I'm ok with leaving it as is for the moment.